python_modules/dagster/dagster/core/execution/run_cancellation_thread.py

```diff
@@ -0,0 +1,42 @@
+import threading
+import time
+from typing import cast
+
+from dagster import check
+from dagster.core.instance import DagsterInstance
+from dagster.core.storage.pipeline_run import PipelineRun, PipelineRunStatus
+from dagster.utils import send_interrupt
+
+INTERVAL_SECONDS = 10
+
+
+def _kill_on_cancel(instance: DagsterInstance, run_id):
+    check.inst_param(instance, "run_id", DagsterInstance)
+    check.str_param(run_id, "run_id")
+
+    while True:
+        time.sleep(INTERVAL_SECONDS)
+        run = cast(
+            PipelineRun,
+            check.inst(
+                instance.get_run_by_id(run_id), PipelineRun, "Run not found for cancellation thread"
+            ),
+        )
+        if run.status in [
+            PipelineRunStatus.CANCELING,
+            PipelineRunStatus.CANCELED,
```

Comment:
are we sure CANCELLED is right here? could that result in this interrupting in a loop?

Reply:
Good question. I think we do want to handle it somehow. Force terminate tries to send a terminate call, this should respond to that too.

I haven't investigated what happens if the process gets multiple interrupts. We could avoid that by having this thread exit after it sends one.